std/algorithm/iteration.d

```diff
@@ -4878,26 +4878,42 @@ private struct UniqResult(alias pred, Range)
 struct Permutations(Range)
     if (isRandomAccessRange!Range && hasLength!Range)
 {
-    size_t[] indices, state;
-    Range r;
+    private size_t[] _indices, _state;
+    private Range _r;
+    private bool _empty;
+
+    // Explicitly undocumented. It will be removed in December 2016. @@@DEPRECATED_2016-12@@@
```

Comment:
> AFAIK Permutations is a very recent addition and the internals of permutations where never publicly documented, that's why I thought just six months would be okay. Imho no deprecation should also be okay, because I can't imagine someone digging through the phobos source code and using the state variable.

Almost one year old. If it hadn't part of a release already it would have been okay, however everything that was (even once) should follow the proper deprecation path.

I cannot imagine someone digging through the source code and accessing Dalvik's internal and modifying the system loader, yet it's done on most of the phones out there.